Enable Tmux support for kitty

meck opened the pull request:
Thanks for a nice app, i work a lot in tmux so i took a stab at enabling serie in kitty
using the "Unicode placeholder" method.

Seems to work fine in kitty, ghostty seems to have some issues with placement. I don't know if
iterm has something equivalent to enable it there as well.

I just wanted se if you are interested in merging and some initial feedback in that case, currently the image tests are failing as the image sizes are now dependent cell size and differs depending on what font is used...

lusingander commented on src/graph/image.rs:
```diff
@@ -134,11 +135,25 @@ pub enum CellWidthType {
+                }
+                CellWidthType::Single => {
+                    (cell_width, cell_height, 1, cell_width / 3, cell_width / 2)
+                }
```
Does dynamically calculating cell sizes have anything to do with this fix in the first place?

meck commented on src/protocol.rs:
```diff
@@ -1,39 +1,70 @@
+        };
     }
     // https://ghostty.org/docs/help/terminfo
     if env::var("TERM").is_ok_and(|t| t == "xterm-ghostty") {
```
No, sorry i meant that is does not work in the sense that the images gets corrupted when using the Unicode Placeholder protocol, but I figured out a solution

meck replied:
Great, ill work on this some more, I think screen should be minimal effort to add as well. zeliji is a no go as of now https://github.com/zellij-org/zellij/issues/3954.

meck commented:
@lusingander I would consider this quite ready and cleaner now, The scope got a bit bigger.

I've rewrote the cell size code to have minimal size when drawing the image and calculate dimensions to match the previous behavior for a 1:2 image.

I discovered that the envars for detecting terminal type might not be present on tmux and ssh session, so i changed the detection algo to query the underlying terminal directly if the kitty protocol is enabled using terminal escape codes.

Ghostty detection now also works as expected, and any future terminals that support the protocol should work without modifications

lusingander commented:
Thank you, I'm not sure if I can check it immediately, but I will.
